src/components/Messages/index.tsx

```diff
@@ -1,26 +1,66 @@
+import { useQuery } from '@apollo/client';
+import Preview from '@components/Messages/Preview';
 import { Card } from '@components/UI/Card';
 import { GridItemEight, GridItemFour, GridLayout } from '@components/UI/GridLayout';
 import MetaTags from '@components/utils/MetaTags';
+import type { Profile } from '@generated/types';
+import { ProfilesDocument } from '@generated/types';
 import isFeatureEnabled from '@lib/isFeatureEnabled';
-import type { Conversation, Stream } from '@xmtp/xmtp-js';
+import type { Conversation, Message } from '@xmtp/xmtp-js';
 import { Client } from '@xmtp/xmtp-js';
-import { useRouter } from 'next/router';
 import type { FC } from 'react';
-import { useEffect, useState } from 'react';
+import { useEffect } from 'react';
 import { APP_NAME } from 'src/constants';
 import Custom404 from 'src/pages/404';
+import Custom500 from 'src/pages/500';
 import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 import { useSigner } from 'wagmi';
 
+export class MessagePreview {
```

Comment:
**suggestion:** class 😕 we avoid using it 🙈

Reply:
Ooooh ok good to know! I thought it might be clean to stitch the Lens & XMTP data together for each item in the list -- but I think I can do it without the new data type 👀